packages/back-end/src/integrations/Databricks.ts

```diff
@@ -0,0 +1,45 @@
+import { DatabricksConnectionParams } from "../../types/integrations/databricks";
+import { runDatabricksQuery } from "../services/databricks";
+import { decryptDataSourceParams } from "../services/datasource";
+import { FormatDialect } from "../util/sql";
+import SqlIntegration from "./SqlIntegration";
+
+export default class Databricks extends SqlIntegration {
+  params!: DatabricksConnectionParams;
+  setParams(encryptedParams: string) {
+    this.params = decryptDataSourceParams<DatabricksConnectionParams>(
+      encryptedParams
+    );
+  }
+  getFormatDialect(): FormatDialect {
+    // sql-formatter doesn't support databricks explicitly yet, so using their generic formatter instead
+    return "sql";
+  }
+  getSensitiveParamKeys(): string[] {
+    const sensitiveKeys: (keyof DatabricksConnectionParams)[] = ["token"];
+    return sensitiveKeys;
+  }
+  runQuery(sql: string) {
+    return runDatabricksQuery(this.params, sql);
+  }
+  toTimestamp(date: Date) {
+    return `TIMESTAMP'${date.toISOString()}'`;
```

Comment:
Since the timestamp is a hardcoded string, using `TIMESTAMP'2022-...'` should be converted to a native timestamp type before it gets executed by the query engine.  The `to_timestamp` method is a lot more powerful since it can convert column values as well as hardcoded strings, so I suspect it's run later and is more expensive.  I haven't actually tested that though, so maybe it doesn't make a difference.

Reply:
sg